rust/arrow/src/array/array_struct.rs

```diff
@@ -69,6 +69,9 @@ impl StructArray {
     }
 
     /// Return child array whose field name equals to column_name
+    ///
+    /// Note: The Arrow specification allows for duplicate field names, and in such
```

Comment:
I didn't find about ` allows for duplicate filed names` from https://github.com/apache/arrow/blob/master/docs/source/format/Columnar.rst#serialization-and-interprocess-communication-ipc

Reply:
There's a duplicate_field_name test, which fails because we always get the first named field, perhaps it hasn't been codified into the spec. I'll check.

Reply:
By allowing duplicate field name is rather strange and uncertain, perhaps that file is used for error checking purpose.

Reply:
From the IPC doc: `A struct is a nested type parameterized by an ordered sequence of types (which can all be distinct), called its fields. Each field must have a UTF8-encoded name, and these field names are part of the type metadata.`

The `which can all be distinct` is totally wasting of time, isn't it?

From C++: https://github.com/apache/arrow/blob/57376d28cf433bed95f19fa44c1e90a780ba54e8/cpp/src/arrow/type.cc

`
    // From this point, there's one or more field in the builder that exists with
    // the same name.

    if (policy_ == CONFLICT_IGNORE) {
      // The ignore policy is more generous when there's duplicate in the builder.
      return Status::OK();
    } else if (policy_ == CONFLICT_ERROR) {
      return Status::Invalid("Duplicate found, policy dictate to treat as an error");
    }
`

From java https://github.com/apache/arrow/blob/25c736d48dc289f457e74d15d05db65f6d539447/java/vector/src/main/java/org/apache/arrow/vector/complex/AbstractStructVector.java

 ```
 /**
   * Policy to determine how to react when duplicate columns are encountered.
   */
  public enum ConflictPolicy {
    // Ignore the conflict and append the field. This is the default behaviour
    CONFLICT_APPEND,
    // Keep the existing field and ignore the newer one.
    CONFLICT_IGNORE,
    // Replace the existing field with the newer one.
    CONFLICT_REPLACE,
    // Refuse the new field and error out.
    CONFLICT_ERROR
  }
```

Reply:
I wasn't aware of this, thanks for finding the references. This could explain why the duplicate_field_names integration tests fail

Reply:
It is not acceptable if we silently append/ignore/replace duplicate fields. Even if we can managed to let user to configure this `global` behavior, it may not satisfy all possible duplications even in a single file: some one want to append but another want to ignore.  So, at present before we can find the solution to let user define the behavior,  I suggest raising error on duplication.

Reply:
That sounds like a good idea, may you please open a JIRA for the work; even if we don't get to complete it in time for 3.0.0.

I also prefer raising an error by default, as that'll make users aware very quickly

Reply:
We had a similar discussion on DataFusion for `RecordBatch` a while back. We now just refuse duplicated column names, as it offers a much, much simpler way of dealing with columns via column names.

However, sometimes people have to parse files with equal column names, which means that arrow somehow has to support them on `RecordBatch`.